### ISSUE TITLE
Customizable (or Semi-Customizable) Alert Email Subjects

### ISSUE THREAD

**Issue by LaikaN57** [kubecost/cost-analyzer-helm-chart#1593]
**What problem are you trying to solve?**

It would be useful if the alert email subject was configurable (or at least partially configurable) so that users can simply filter on subject lines for different alert emails they might get from Kubecost. The thought here is when setting up multiple email reports, the user should get alert emails with subjects indicating which report the email is linked to without having to view the message body.

Right now the alert email headers look somewhat like this:

```
Date: Wed, 27 Jul 2022 20:54:30 +0000 (UTC)
From: Kubecost <notify@kubecost.com>
Mime-Version: 1.0
Message-ID: <[REDACTED]>
Subject: Kubecost Alert
```

**Describe the solution you'd like**

It would be better to have alerts setup such that they can be easily filtered by subject. For instance if you have multiple reports. An example would be maybe you have a report by namespace and a report by cluster. These should have subject headers like:

- `Subject: [Kubecost Alert] My Report Title for Namespaces`
- `Subject: [Kubecost Alert] My Report Title for Clusters`

Email filtering by subject as opposed to a "full text search" allows for search rules with 1 term like:

`Subject == "[Kubecost Alert] My Report Title for Namespaces"`

vs 2 terms like:

`From == notify@kubecost.com AND Message contains "My Report Title for Namespaces"`

**Describe alternatives you've considered**

None

**How would users interact with this feature?**

Users would primary interact with this feature in 2 places:

1. The alerts page in the Kubecost UI
2. The user's email inbox

They would be able to create, edit, and view alert titles from the Kubecost UI. They would then be able to setup filtering on their email inbox to search or sort Kubecost report alerts to their liking.

gz#2264<br><br><i>(related to <a href='https://kubecost.zendesk.com/agent/tickets/2264'>Zendesk ticket #2264</a>)</i>

**Comment by LaikaN57** [comment]
NOTE: I realized that I was seeing this on "Test Alerts". I have not seen the subject of a real email report alert. Looking for others to confirm. I will also attempt to confirm this as well and will post an update or close.

**Comment by kirbsauce** [comment]
hi @LaikaN57 , thanks for the suggestion!  I too do a lot of filters on emails and can definitely see the value.  Stay tuned for any followup questions from our product team!

**Comment by dwbrown2** [comment]
+1 on this! Might just need to think through handling special characters and other edge cases.

**Comment by LaikaN57** [comment]
I'm pretty sure email subjects mostly support utf-8 now days (as proved out by all the scammy emails using "cute" emojis in my inbox).

**Comment by LaikaN57** [comment]
Confirmed that alerts generated by reports do not contain report name or similar. Proof:
```
Content-Type: multipart/alternative; boundary=d7e5eb547d4c9d86c95325bb51897a7f4a0c7a1a5f4a971eb7736c813988
Date: Fri, 29 Jul 2022 00:00:00 +0000 (UTC)
From: Kubecost <notify@kubecost.com>
Mime-Version: 1.0
Message-ID: <UJ0y4Ag3QGKSpZK-4TCN7A@geopod-ismtpd-6-5>
Subject: Kubecost Alert

--d7e5eb547d4c9d86c95325bb51897a7f4a0c7a1a5f4a971eb7736c813988
Content-Transfer-Encoding: quoted-printable
Content-Type: text/plain; charset="UTF-8"
Mime-Version: 1.0

Below is the Kubecost daily summary for your cumulative spend by namespace.
The summary is from 2022-07-28 to 2022-07-29.
```

**Comment by akramwaseem** [comment]
+1

**Comment by dwbrown2** [comment]
Goal is to review + get this in to 1.98!

**Comment by danielfritsch** [comment]
+1 on this request! Would be awesome to be able to customize the subject name in the emails to include which cluster the report is about -- very useful when you are managing multiple Kubecost instances for a multi-cluster setup

**Comment by Adam-Stack-PM** [comment]
@danielfritsch Thanks for  the +1 and building on the idea. 

This work will not make it in v1.98 but we are planning a big push on alerts and notifications for an upcoming release.

**Comment by dwbrown2** [comment]
Specifically, 1.99 is current plan!

**Comment by kirbsauce** [comment]
@Adam-Stack-PM , @dwbrown2 will the customization also extend to the email body?

**Comment by dwbrown2** [comment]
@kirbsauce any example scenarios you have in mind?

**Comment by kirbsauce** [comment]
No use case, just curious. That said, one user did want the cluster name/id but I suspect having it in the subject will work. I'm currently trying to confirm this with them. 

Working with automations in other tools with email notifications, I've found it very useful to embed more robust information in the email body, which triggered my question.

**Comment by kirbsauce** [comment]
@dwbrown2 , confirmed with user that Alert Name and Cluster ID in subject would work.

**Comment by dwbrown2** [comment]
@AjayTripathy was able to pick this up. Backend PR currently outstanding. Currently expected in 1.99. 

@AjayTripathy can you share info on parameters supported?

cc @teevans would now just need a minor FE change.

**Comment by teevans** [comment]
@nealormsbee - Mind taking a look at what needs to be hammered out here?

**Comment by nealormsbee** [comment]
Sure thing! It seems like we'd just need a per-alert way to set a subject line

**Comment by AjayTripathy** [comment]
@nealormsbee https://github.com/kubecost/kubecost-emailer/pull/3 PR here

**Comment by aibarbetta** [comment]
is this available in v1.100.1? I see the global subject can be changed from the Kubecost UI, but I'd like to configure it from Helm

**Comment by nealormsbee** [comment]
Hi @aibarbetta !

You can indeed customize the global alert subject through Helm using `.Values.global.notifications.globalAlertEmailSubject`.
Thanks to your question, I've opened a PR to show this option as a comment in our `values.yaml` file: https://github.com/kubecost/cost-analyzer-helm-chart/pull/2005

Cheers.
